File: tests/java/com/trulioo/sdk/model/CommunicationTest.java
package com.trulioo.sdk.model;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.*;

public class CommunicationTest {
    private Communication communication;

    @BeforeEach
    public void beforeEach() {
        communication = new Communication();
    }

    @Test
    public void testMobileNumber() {
        String mobileNumber = "18887730179";
        communication.setMobileNumber(mobileNumber);

        assertEquals(mobileNumber, communication.getMobileNumber());
        assertEquals(mobileNumber, new Communication().mobileNumber(mobileNumber).getMobileNumber());
    }

    @Test
    public void testTelephone() {
        String telephone = "18887730179";
        communication.setTelephone(telephone);

        assertEquals(telephone, communication.getTelephone());
        assertEquals(telephone, new Communication().telephone(telephone).getTelephone());
    }

    @Test
    public void testTelephone2() {
        String telephone2 = "18887730179";
        communication.setTelephone2(telephone2);

        assertEquals(telephone2, communication.getTelephone2());
        assertEquals(telephone2, new Communication().telephone2(telephone2).getTelephone2());
    }

    @Test
    public void testEmailAddress() {
        String emailAddress = "support@trulioo.com";
        communication.setEmailAddress(emailAddress);

        assertEquals(emailAddress, communication.getEmailAddress());
        assertEquals(emailAddress, new Communication().emailAddress(emailAddress).getEmailAddress());
    }

    @Test
    public void testEquals() {
        String mobileNumber = "18887730179";
        Communication communication1 = new Communication().mobileNumber(mobileNumber);

        assertEquals(communication1, communication1);
        assertEquals(communication1, new Communication().mobileNumber(mobileNumber));
        assertNotEquals(communication1, new Communication().mobileNumber(mobileNumber + "1"));
        assertNotEquals(communication1, null);
    }

    @Test
    public void testHashcode() {
        String mobileNumber = "18887730179";

        assertEquals(new Communication().mobileNumber(mobileNumber).hashCode(),
                new Communication().mobileNumber(mobileNumber).hashCode());
    }

    @Test
    public void testToString() {
        String mobileNumber = "18887730179";

        assertEquals(new Communication().mobileNumber(mobileNumber).toString(),
                new Communication().mobileNumber(mobileNumber).toString());
    }
}

File: tests/unit/php/Model/CommunicationTest.php
<?php

namespace Trulioo\SDK\Test\Model;

use InvalidArgumentException;
use PHPUnit\Framework\TestCase;
use Trulioo\SDK\Model\Communication;

class CommunicationTest extends TestCase
{
    private Communication $communication;

    public function setUp(): void
    {
        $this->communication = new Communication([]);
    }

    public function testPropertyMobileNumber()
    {
        $this->communication->setMobileNumber('18887730179');
        $this->assertEquals('18887730179', $this->communication->getMobileNumber());

        try {
            $this->communication->setMobileNumber(str_repeat('1', 201));
            $this->fail('Expected exception');
        } catch (InvalidArgumentException $e) {
        }
    }

    public function testPropertyTelephone()
    {
        $this->communication->setTelephone('18887730179');
        $this->assertEquals('18887730179', $this->communication->getTelephone());

        try {
            $this->communication->setTelephone(str_repeat('1', 201));
            $this->fail('Expected exception');
        } catch (InvalidArgumentException $e) {
        }
    }

    public function testPropertyTelephone2()
    {
        $this->communication->setTelephone2('18887730179');
        $this->assertEquals('18887730179', $this->communication->getTelephone2());

        try {
            $this->communication->setTelephone2(str_repeat('1', 201));
            $this->fail('Expected exception');
        } catch (InvalidArgumentException $e) {
        }
    }

    public function testPropertyEmailAddress()
    {
        $this->communication->setEmailAddress('support@trulioo.com');
        $this->assertEquals('support@trulioo.com', $this->communication->getEmailAddress());

        try {
            $this->communication->setEmailAddress(str_repeat('a', 201));
            $this->fail('Expected exception');
        } catch (InvalidArgumentException $e) {
        }
    }
}
